l10n_it_website_sale_fiscalcode/controllers/main.py

```diff
@@ -1,24 +1,39 @@
 # -*- coding: utf-8 -*-
 # Copyright 2017 Nicola Malcontenti - Agile Business Group
+# Copyright 2019 Simone Rubino
 # License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
 
+from odoo import _
 from odoo.http import request
 from odoo.addons.website_sale.controllers.main import WebsiteSale
 
 
 class WebsiteSaleFiscalCode(WebsiteSale):
 
-    def _get_mandatory_billing_fields(self):
-        res = super(
-            WebsiteSaleFiscalCode, self
-        )._get_mandatory_billing_fields()
-        res.append('fiscalcode')
-        return res
-
     def _checkout_form_save(self, mode, checkout, all_values):
         res = super(WebsiteSaleFiscalCode, self)._checkout_form_save(
             mode, checkout, all_values)
+        partner_values = dict()
         if 'fiscalcode' not in checkout and 'fiscalcode' in all_values:
-            request.env['res.partner'].browse(res).sudo().write(
-                {'fiscalcode': all_values['fiscalcode']})
+            partner_values['fiscalcode'] = all_values['fiscalcode']
+        if 'is_company' not in checkout and 'is_company' in all_values:
```

Comment:
@monen17 e se `is_company` è in `checkout`?

Reply:
Eventualmente aggiungi un commento per spiegarlo

Reply:
Ah ma vedo che anche per `fiscalcode` viene fatto lo stesso.
Comunque se sai rispondere metti pure il commento, altrimenti procediamo.

Reply:
Aggiunto il commento